### PR TITLE
[DA-100] Distinguish when GA was not found in AProx and when the comm…

### PR DESCRIPTION
**Pull request by janinko**
…unication failed

When GA is not found DA will return 404 or 206 codes.
When communication fails DA will return 502 code.

### CODE REVIEW

**Comment by janinko** [comment]
There is problem in AProx: https://github.com/Commonjava/aprox/issues/207

**Comment by jdcasey** [comment]
FWIW, I think traditionally 502 or 503 would be a better response. It lets the caller know it's a communication issue with a remote service on the backend.

**Comment by janinko** [comment]
I updated the code from 500 to 502 as jdcasey suggested. But there is still the issue in AProx, until it is resolved and deployed, the test will fail. So until then let this PR :zzz:

**Comment by jdcasey** [comment]
I've deployed a new instance of AProx on the old NCL VM (10.3.8.115:80) with this fix and the new depgraph api.

**Comment by jbartece** [comment]
retest this please

File: reports-rest/src/main/java/org/jboss/da/rest/reports/Reports.java [hunk 1]
```diff
@@ -102,44 +105,41 @@ public Response gavGenerator(@ApiParam(
...
-            } catch (CommunicationException e) {
-                responseStatus = 206; // Partial Content HTTP code
+                VersionLookupResult lookupResult = versionFinder.lookupBuiltVersions(gav);
+                LookupReport lookupReport = toLookupReport(gav, lookupResult);
+                reportsList.add(lookupReport);
+                if (lookupResult == null) {
+                    responseStatus = Status.PARTIAL_CONTENT;
+                }
+            } catch (CommunicationException ex) {
+                return Response.status(502).entity(ex).build();
```
Comment (jbartece) [inline reports-rest/src/main/java/org/jboss/da/rest/reports/Reports.java]: Probably we should return JSON formatted response.

**Comment by jbartece** [comment]
:+1:

**Pull request merged**